src/bootstrap/config.rs

```diff
@@ -847,7 +845,11 @@ impl Config {
         set(&mut config.compiler_docs, build.compiler_docs);
         set(&mut config.docs_minification, build.docs_minification);
         set(&mut config.docs, build.docs);
-        set(&mut config.fast_submodules, build.fast_submodules);
+        if build.fast_submodules.is_some() {
+            println!(
+                "warning: the `fast_submodules` option is deprecated and does nothing. Fast submodules are enabled unconditionally."
+            );
+        }
```

Comment:
It doesn't seem like there's a significant advantage in continuing to accept the option if we're removing it, a hard error seems reasonable to me.

Reply:
Sure thing, done. I also added the removal (but didn't mark it as a "major change" since I don't think anyone is actually using it).